### PR TITLE
fix(activate): restore the `--` separator PowerShell drops in an activated pwsh session

### PR DESCRIPTION
**Pull request by jdx**
In a shell running `mise activate pwsh`, `mise exec -- pnpm --version` failed:

```
error: unexpected argument '--version' found
  tip: a similar argument exists: '--verbose'
Usage: mise exec [FLAGS] [TOOL@VERSION]… [-- COMMAND]…
```

The same command run through `mise.exe` directly printed the version. This PR makes
the activated form work too.

## Why it happened

`mise activate pwsh` defines `mise` as a PowerShell function, and PowerShell's
parameter binder claims the first bare `--` on a line as its own end-of-parameters
token. It removes that token before `$args` is populated, so the wrapper forwarded
`exec pnpm --version` and mise read `--version` as its own flag.

Nothing about mise is involved — the binder does it to any function:

```powershell
PS> function f { $args -join '|' }
PS> f exec -- pnpm --version
exec|pnpm|--version
```

It takes exactly one separator, wherever it sits: `f a b -- c -- d` yields
`a|b|c|--|d`. No spelling of the function signature avoids it — `$args`,
`[Parameter(ValueFromRemainingArguments)]` and `PositionalBinding=$false` all lose
it identically. bash, zsh and fish forward `"$@"` untouched, which is why only pwsh
needed a fix.

The failure was quiet until it wasn't: the separator only matters once the command
being run has a flag that collides with one of mise's own, so `mise exec -- pnpm install`
worked and `mise exec -- pnpm --version` did not.

## What changed

The raw invocation text still holds the separator. The wrapper now parses
`$MyInvocation.Line`, locates its own invocation by `$MyInvocation.OffsetInLine`, and
reinserts `--` at the position the binder took it from:

```powershell
PS> mise exec -- pnpm --version    # activated
11.25.0
PS> mise x node@20 -- node --version
v20.19.0
PS> mise ls --offline | Out-Null; mise exec -- pnpm --version   # both repaired
11.25.0
```

Guards on the repair:

- A regex checks for a bare `--` on the line before anything is parsed, so ordinary
  calls (`mise ls`, `mise install`, the prompt's `hook-env`) cost nothing extra.
- An already-quoted `mise exec '--' pnpm --version`, which the binder never touched,
  is left alone rather than given a second separator.
- Only the elements *before* the separator decide where it goes, since each contributes
  exactly one argument. What follows may expand freely, so `mise exec -- pnpm @childArgs`
  is repaired too. A splat *before* the separator hides how many arguments precede it, and
  there the repair stands down rather than guess — that line behaves as it does today.

## Scope

Affected are the commands where `--` is required: `mise exec` / `mise x`,
`mise tasks add`, `mise dotfiles capture`, `mise oci run`,
`mise generate git-pre-commit` and `mise bootstrap`. The fix sits in the wrapper, so
it covers all of them.

`mise run` was **not** affected and is unchanged: it takes `double_dash = "automatic"`
with `unknown_flags = "value"`, so flags after the task name reach the task whether or
not a separator survives. `mise run t -- --jobs 4`, `--quiet`, `--raw` and the bare
`mise <task> --flag` shorthand all behave identically with and without the `--`.

## Adoption

- An already-open PowerShell session keeps the old wrapper until `mise activate pwsh`
  runs again, normally at the next shell start.
- `mise exec -- -- pnpm --version` — the doubled form people arrive at on their own —
  now fails in an activated shell, the same way it already fails without activation
  (`"--" couldn't exec process`). That is the intended consistency, but anyone who
  scripted the doubled form should drop back to a single `--`.
- No new flags or settings, and no change outside pwsh.

## Validation

`e2e-win/activate_double_dash.Tests.ps1` covers `exec`, the `x` alias, the quoted
separator, two invocations sharing a line, a splat on either side of the separator, a
lone argument, and a command with no separator at all. Each case activates in a child
`pwsh -NoProfile` so a developer profile or another suite's in-process activation
cannot decide the result.

Run under Linux pwsh 7.6.6 locally: 8/8 pass against this build, and as a negative
control 3 of them fail against an unfixed `mise` — so the tests discriminate rather
than passing by construction. Unit tests for `src/shell/pwsh.rs` pass (17/17,
activation and deactivation snapshots updated) and `mise run lint` is clean.

The behavior was reproduced on pwsh 7.6.6 on Linux as well as the reporter's Windows
PowerShell 5.1 and pwsh 7.6.5, so it is a property of the binder rather than of
Windows. CI's Windows lane runs pwsh 7 only; the constructs used
(`Parser::ParseInput`, `Ast.Find`, `[System.Collections.ArrayList]::new()`) are all
available in Windows PowerShell 5.0+, but 5.1 is not covered by an automated test.

Reported in #13196.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

*AI-assisted — Tool: Claude Code; model: anthropic/claude-opus-5; version: 2.1.270.*

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed PowerShell activation handling for the `--` argument separator when using `mise exec` and `mise x`.
  * Improved handling of quoted separators, multiple commands, splatted arguments, and single arguments.
  * Commands without a separator continue to work unchanged.
  * PowerShell activation now correctly handles unsupported `-NoProfile` placement.
* **Tests**
  * Added end-to-end coverage for separator forwarding, activation behavior, and related edge cases.

---

> [!NOTE]
> **Medium Risk**
> Every activated pwsh `mise` call now runs through AST-based argument repair; incorrect edge-case handling could mis-forward args, though scope is pwsh-only and heavily tested.
> 
> **Overview**
> Fixes **`mise exec` / `mise x`** (and other `--`-separated subcommands) in an activated **pwsh** session: PowerShell strips the first bare `--` before the `mise` function sees `$args`, so commands like `mise exec -- pnpm --version` were forwarded without the separator and mise treated downstream flags as its own.
> 
> Activation now emits **`__mise_restore_double_dash`**, which reads `$MyInvocation.Line`, finds this invocation via the AST, and reinserts `--` at the correct index (with a fast regex skip when no bare `--` is present). The wrapper passes repaired args through; **deactivate** removes the helper. Guards avoid double-inserting quoted `'--'`, skip repair when a splat precedes the separator, and use `return ,$array` so a single argument is not unrolled into a string.
> 
> **Tests:** new Windows e2e suite (`e2e-win/activate_double_dash.Tests.ps1`) plus Rust unit/snapshot updates for activate and deactivate output.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit aa28e7718792583ea00715f70201ca7e14654794. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/shell/pwsh.rs [hunk 1]
```diff
@@ -33,8 +33,64 @@ impl Shell for Pwsh {
...
+                $index = -1
+                for ($i = 1; $i -lt $elements.Count; $i++) {{
+                    # Extent.Text is the source spelling, so a quoted '--' -- which the binder
+                    # leaves alone and which is therefore already in $Arguments -- does not match.
+                    if ($elements[$i].Extent.Text -eq '--') {{ $index = $i; break }}
+                }}
+                if ($index -lt 0) {{ return ,$Arguments }}
+                # The element list maps one-to-one onto $Arguments, minus the command name and
+                # the separator the binder took. When it does not -- a splatted @array, say --
+                # the position cannot be trusted, so leave the arguments alone.
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/shell/pwsh.rs]: Valid — fixed in b6ccee191.

The guard was conservative in a direction it didn't need to be. Only what stands *before* the separator fixes where it goes: each element there contributes exactly one argument, so the insertion index is that offset regardless of how the tail expands. Comparing the whole element list against `$args` threw that away.

It now scans only the leading elements, and stops at a splat *before* the separator — the one case where the position genuinely isn't knowable. Verified in pwsh 7.6.6:

```
mise exec -- pnpm @childArgs       -> exec|--|pnpm|--version|--silent   (repaired)
mise exec node@20 -- pnpm @rest    -> exec|node@20|--|pnpm|--version    (repaired)
mise exec @pre -- node --version   -> unchanged                          (stands down)
```

Both shapes are now pinned in `e2e-win/activate_double_dash.Tests.ps1`.

*AI-assisted — Tool: Claude Code; model: anthropic/claude-opus-5; version: 2.1.270.*

File: e2e-win/activate_double_dash.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    # genuinely is not knowable and the repair has to stand down rather than guess.
+    It 'leaves a separator preceded by a splatted array alone' {
+        $body = @(
+            "`$pre = @('-C', '.')"
+            "mise exec @pre -- pwsh -NoProfile -File '$script:echoArgs' --version"
+        ) -join "`n"
+        $out = Invoke-Activated $body $script:config
+
+        $out | Should -Not -Match 'ARGS='
+    }
```
[the remark this answers is not in the document]
Comment (jdx) [reply e2e-win/activate_double_dash.Tests.ps1:105]: Fair — that assertion passed vacuously. Fixed in aa28e7718: it now rejects `ACTIVATION-ERROR` and asserts the failure mise actually produces (`unexpected argument '-NoProfile'`, the first word past where the `--` used to be), so the test proves the stand-down path ran rather than merely that nothing did.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

*AI-assisted — Tool: Claude Code; model: anthropic/claude-opus-5; version: 2.1.270.*
